Horatio/Operations/AlertOperation.swift

```diff
@@ -64,24 +64,39 @@ class AlertOperation: Operation {
     }
 
     override func execute() {
-        var presentationViewController: UIViewController?
-        
-        if let presentationContext = presentationContext {
-            presentationViewController = presentationContext
-        } else {
-            presentationViewController = UIApplication.shared.keyWindow?.rootViewController
+        // it's probably not safe to even walk the view hierarchy from a background thread, so do this all on main
+        DispatchQueue.main.async {
```

Comment:
I think it helps ensure nothing's changing in the view hierarchy while we're waiting (it doesn't guarantee it, but it does significantly lessen the possibility). Nothing on main thread _should_ be blocking on this operation, so it should be allowed to assume it can main a few statements.

That said, I tend to agree about being wary of deadlocks, so I leave it up to you to decide.